### PR TITLE
Update Minio chart and fix GCS Gateway mode

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Updated Minio chart
- Fixed GCS Gateway mode

**Related issue(s)**

#3612 

See also kyma-project/test-infra#902

### CODE REVIEW

File: resources/assetstore/charts/asset-store-controller-manager/templates/statefulset.yaml [hunk 1]
```diff
@@ -45,10 +45,7 @@ spec:
                 name: assetstore-minio-docs-upload
                 key: APP_UPLOAD_ENDPOINT_WITH_PORT
           - name: APP_STORE_EXTERNAL_ENDPOINT
-            valueFrom:
-              configMapKeyRef:
-                name: assetstore-minio-docs-upload
-                key: APP_EXTERNAL_ENDPOINT
+            value: {{ if .Values.global.minioExternalEndpoint }}{{ .Values.global.minioExternalEndpoint }}{{ else }}'https://minio.{{ .Values.global.ingress.domainName }}'{{ end }}
```
Comment (magicmatatjahu) [inline resources/assetstore/charts/asset-store-controller-manager/templates/statefulset.yaml]: If you use global variables in this condition and in various places in asset-store charts, maybe would be better to extract this logic to global values as storeExternalEndpoint?

File: resources/assetstore/charts/minio/templates/content-upload-configmap.yaml [hunk 2]
```diff
@@ -13,6 +13,6 @@ data:
   APP_UPLOAD_ENDPOINT_WITH_PORT: '{{ template "minio.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.service.port }}'
   APP_BUCKET_NAME: content
   APP_BUCKET_REGION: us-east-1
-  APP_EXTERNAL_ENDPOINT: https://minio.{{ .Values.global.ingress.domainName }}
+  APP_EXTERNAL_ENDPOINT: {{ if .Values.global.minioExternalEndpoint }}{{ .Values.global.minioExternalEndpoint }}{{ else }}'https://minio.{{ .Values.global.ingress.domainName }}'{{ end }}
```
Comment (magicmatatjahu) [inline resources/assetstore/charts/minio/templates/content-upload-configmap.yaml]: If you use global variables in this condition and in various places in asset-store charts, maybe would be better to extract this logic to global values as storeExternalEndpoint?

File: resources/assetstore/charts/asset-upload-service/templates/deployment.yaml [hunk 3]
```diff
@@ -54,10 +54,7 @@ spec:
                   name: assetstore-minio
                   key: accesskey
             - name: APP_UPLOAD_EXTERNAL_ENDPOINT
-              valueFrom:
-                configMapKeyRef:
-                  name: assetstore-minio-docs-upload
-                  key: APP_EXTERNAL_ENDPOINT
+              value: {{ if .Values.global.minioExternalEndpoint }}{{ .Values.global.minioExternalEndpoint }}{{ else }}'https://minio.{{ .Values.global.ingress.domainName }}'{{ end }}
```
Comment (magicmatatjahu) [inline resources/assetstore/charts/asset-upload-service/templates/deployment.yaml]: If you use global variables in this condition and in various places in asset-store charts, maybe would be better to extract this logic to global values as storeExternalEndpoint?

File: resources/assetstore/charts/minio/templates/deployment.yaml [hunk 4]
```diff
@@ -37,15 +40,17 @@ spec:
...
-      # MODIFIED: Block - START
-      {{- if .Values.deployment.annotations }}
-      annotations:
-{{ toYaml .Values.deployment.annotations | indent 8 }}
-      {{- end }}
-      # MODIFIED: Block - END
       labels:
         app: {{ template "minio.name" . }}
         release: {{ .Release.Name }}
+      {{- if .Values.podAnnotations }}
```
Comment (magicmatatjahu) [inline resources/assetstore/charts/minio/templates/deployment.yaml]: Extract this condition to global? You also use this logic in statefulset.

File: resources/assetstore/charts/asset-store-controller-manager/templates/statefulset.yaml (hunk 1 above)
Comment (michal-hudy) [reply resources/assetstore/charts/asset-store-controller-manager/templates/statefulset.yaml]: `values.yaml` may contain only values, logic is not supported.

File: resources/assetstore/charts/minio/templates/content-upload-configmap.yaml (hunk 2 above)
Comment (michal-hudy) [reply resources/assetstore/charts/minio/templates/content-upload-configmap.yaml]: `values.yaml` may contain only values, logic is not supported.

File: resources/assetstore/charts/asset-upload-service/templates/deployment.yaml (hunk 3 above)
Comment (michal-hudy) [reply resources/assetstore/charts/asset-upload-service/templates/deployment.yaml]: `values.yaml` may contain only values, logic is not supported.

File: resources/assetstore/charts/minio/templates/deployment.yaml (hunk 4 above)
Comment (michal-hudy) [reply resources/assetstore/charts/minio/templates/deployment.yaml]: `values.yaml` may contain only values, logic is not supported.

**Comment by michal-hudy** [comment]
`pre-master-kyma-gke-minio-gateway` can be ignored as it will not work until full migration to the Asset Store.

**Pull request merged**